src/utils/dateTimeFormatter.ts

```diff
@@ -1,7 +1,21 @@
-import { format } from "date-fns";
+import { format, formatDistance } from "date-fns";
+import { zhCN } from "date-fns/locale";
+import i18n from "i18n-js";
 
 export const formatDateTime = (date: number | Date): string =>
   format(date, "d MMM yyyy, h:mma");
 
 export const formatDate = (date: number | Date): string =>
   format(date, "d MMM yyyy");
+
+export const formatTimeDifference = (
+  start: number | Date,
+  end: number | Date
+): string => {
+  if (i18n.locale.includes("zh")) {
```

Comment:
Does this imply there can be multiple locales? What happens if there are both `en` and `zh`?

Reply:
to set a locale accepted by the date-fns we need to import it, the format of the locale is zhCN while our expo locale detection reflects "zh-SG'/"zh-US". by default, date-fns uses the english version so no checks required

Reply:
Actually I was more asking what is in `i18n.locale` such that we need to use `.includes`. Is it an array of locales?

Reply:
the i18n.locale is a string of eg  "zh-SG" (chinese lang, SG location) or "zh-US" (chinese lang, US location) so the .includes check is to check if a substring of "zh" is inside so the date formatter knows it needs to use the zhCN library

Reply:
I see, what if we do `startsWith` instead so we don't randomly match to something like `xxxzhyyy`

Reply:
Good idea😄!